sda-commons-cloudevents/src/test/java/org/sdase/commons/server/cloudevents/ContractCloudEventTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.server.cloudevents;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.*;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.sdase.commons.server.cloudevents.app.KafkaTestApp;
+import org.sdase.commons.server.cloudevents.app.contract.ContractCreatedEvent;
+import org.sdase.commons.server.cloudevents.app.contract.ContractCreatedMessageProducer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.KafkaMessageListenerContainer;
+import org.springframework.kafka.listener.MessageListener;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.ContainerTestUtils;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.annotation.DirtiesContext;
+
+@SpringBootTest(
+    classes = KafkaTestApp.class,
+    properties = {
+      "management.server.port=0",
+      "spring.kafka.bootstrap-servers=${spring.embedded.kafka.brokers}",
+      "spring.kafka.producer.value-serializer=org.springframework.kafka.support.serializer.JsonSerializer"
+    })
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {"listeners=PLAINTEXT://localhost:9092", "port=9092"})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
```

Comment:
Do you know the reason why we use `DirtiesContext` here? Does every Kafka test need/use that annotation?

Reply:
My understanding is that the EmbeddedKafka annotation adds an embedded Kafka to the test context which is cleaned up using DirtiesContext. used in the example provided by the spring boot kafka [documentation](https://docs.spring.io/spring-kafka/reference/html/#embedded-kafka-annotation) If not used there is a chance that some spring caching magic uses the same kafka broker in other tests.

Actually i think that both Kafka tests in cloud events could use the same broker which would speed up test execution. The documentation provides an [example](https://docs.spring.io/spring-kafka/reference/html/#same-broker-multiple-tests), although it would be easier to just put the two test in the same test class.

However, these test provide an easy to understand and safe example that will be most likely used in other tests